src/action/stat.rs

```diff
@@ -0,0 +1,489 @@
+            warn!("Unable to get extended attributes: {}", err);
+            return vec![]
+        }
+    };
+
+    let mut result = vec![];
+    for attr in xattrs {
+        match xattr::get(path, &attr) {
+            Ok(attr_value) => result.push(ExtAttr {
+                name: attr.into_vec(),
+                value: attr_value.unwrap_or_default(),
+            }),
+
+            Err(err) => warn!("Unable to get an extended attribute: {}", err),
+        }
+    }
+    result
+}
+
+#[cfg(not(target_family = "unix"))]
+fn get_ext_attrs(path: &Path) -> Vec<ExtAttr> {
+    vec![]
+}
+
+fn get_time_option<E: std::fmt::Display>(time: Result<SystemTime, E>) -> Option<SystemTime> {
+    match time {
+        Ok(time_value) => Some(time_value),
+        Err(err) => {
+            warn!("Unable to get time value: {}", err);
+            None
+        }
+    }
+}
+
+fn get_time_since_unix_epoch(sys_time: &Option<SystemTime>) -> Option<u64> {
+    match sys_time {
+        Some(time_value) => match rrg_proto::micros(*time_value) {
+            Ok(micros_value) => Some(micros_value),
+            Err(error) => {
+                warn!("failed to convert time: {}", error);
+                None
+            }
+        }
+        None => None
+    }
+}
+
+#[cfg(target_os = "linux")]
+fn get_status_change_time(metadata: &Metadata) -> Option<SystemTime> {
+    use std::time::Duration;
+    use std::os::unix::fs::MetadataExt;
+
+    UNIX_EPOCH.checked_add(Duration::from_secs(metadata.ctime() as u64))
+}
+
+#[cfg(target_os = "linux")]
+fn form_response(original_path: &Path, destination: &Path)
+                 -> Result<Response, std::io::Error> {
+    use std::os::unix::fs::MetadataExt;
+
+    let metadata = fs::symlink_metadata(destination)?;
+    let original_metadata = fs::symlink_metadata(original_path)?;
+
+    Ok(Response {
+        mode: metadata.mode() as u64,
+        inode: metadata.ino() as u32,
+        device: metadata.dev() as u32,
+        hard_links: metadata.nlink() as u32,
+        uid: metadata.uid() as u32,
+        gid: metadata.gid() as u32,
+        size: metadata.size() as u64,
+        access_time: get_time_option(metadata.accessed()),
+        modification_time: get_time_option(metadata.modified()),
+        status_change_time: get_status_change_time(&metadata),
+        block_count: metadata.blocks() as u32,
+        block_size: metadata.blksize() as u32,
+        represented_device: metadata.rdev() as u32,
+        flags_linux: get_linux_flags(destination),
+
+        symlink: match original_metadata.file_type().is_symlink() {
+            true => Some(fs::read_link(original_path).unwrap()),
+            false => None
+        },
+
+        path: original_path.to_owned(),
+
+        extended_attributes: vec![],
+    })
+}
+
+#[cfg(not(target_os = "linux"))]
+fn form_response(original_path: &PathBuf, destination: &PathBuf)
+                 -> Result<Response, std::io::Error> {
+    Ok(Response::default())
```

Comment:
Looks like this lines prevents the code from compiling on Windows and macOS. Since you don't support these platforms, why not just return `Err` here?

Reply:
[Fixed](https://github.com/google/rrg/pull/34/commits/f19a8b879643b4dde2948d53a713ca5aa48cd12d)